python/ray/autoscaler/gcp/node_provider.py

```diff
@@ -43,7 +44,12 @@ def __init__(self, provider_config, cluster_name):
         NodeProvider.__init__(self, provider_config, cluster_name)
 
         self.lock = RLock()
-        self.compute = discovery.build("compute", "v1")
+        service_account_info = provider_config.get("gcp_credentials")
+        gcp_credentials = service_account.Credentials.from_service_account_info(
```

Comment:
To make sure, this still works if gcp credentials is empty right (as before)?

Reply:
Also does it make sense to pull out `credentials_from_config` so we can dedup between node_provider & config

Reply:
Sure, since node_provider.py already imports from config.py, does it make sense to put this function in config.py?